gdnative-core/src/core_types/geom/transform.rs

```diff
@@ -87,6 +87,9 @@ impl Transform {
     }
 
     /// Returns this transform, with its origin moved by a certain `translation`
+    #[deprecated = "`translated` is not relative to the transform's coordinate system \
+    and thus inconsistent with GDScript. Please use translated2() instead. \
+    This method will be removed in gdnative 0.11."]
```

Comment:
I haven't enough time to take a closer look, but isn't that exactly what we wanted to fix for a very long time in https://github.com/godotengine/godot/pull/55923

Currently GDScript definitely has a major inconsistency between `translated` on the one hand and `rotated` + `scaled` on the other hand, see https://github.com/godotengine/godot-proposals/issues/1336. Basically everyone in the discussion is on the same page that we want to resolve it, but the direction is which to resolve was discussed to death, which is why the PR is stuck at the moment. I still hope to get it resolved one way or the other, so I'm not sure if it is very crucial to stick to the status quo of GDScript in this case due to the inconsistency.

Reply:
@bluenote10 I even upvoted that issue a while ago, thanks a lot for reminding me in this context! 

I'm totally with you in that case -- we should probably keep the current behavior and not deprecate anything. However, this discrepancy from GDScript (at least in Godot 3) should be documented.

What do you think about adding a method `transform_global` which implements the current GDScript way (would that even be the right name?)

Reply:
> What do you think about adding a method transform_global which implements the current GDScript way (would that even be the right name?)

I think it was the other way around: `scaled` + `rotated` + `translated` should have global semantics (and no suffix in the name) and the local variants all get a `_local` suffix, because that's how `Basis` already does it. But I'm a bit short on time and without actually looking into it again I may be getting it wrong now ;) For the desired target state I'd refer to code in https://github.com/godotengine/godot/pull/55923

Reply:
Awesome, https://github.com/godotengine/godot/pull/55923 has now been merged! 🚀